source/server/session_utilities_service.cpp

```diff
@@ -50,8 +52,37 @@ ::grpc::Status SessionUtilitiesService::ResetServer(::grpc::ServerContext* conte
   }
   bool is_server_reset = session_repository_->reset_server();
   response->set_is_server_reset(is_server_reset);
-  device_enumerator_->clear_syscfg_session();
+
+  {
+    std::unique_lock<std::shared_mutex> lock(observers_mutex_);
+    auto i = observers_.begin();
+    auto e = observers_.end();
+    for (; i != e; ++i)
```

Comment:
Could you use `for (auto observer : observers_)` here?

Reply:
Updated as suggested.